pkg/network/deviceinfo/sriov.go

```diff
@@ -57,45 +55,30 @@ func CreateNetworkPCIAnnotationValue(networks []v1.Network, interfaces []v1.Inte
 
 func mapNetworkNameToPCIAddress(networks []v1.Network, interfaces []v1.Interface,
 	networkStatusAnnotationValue string) (map[string]string, error) {
-	multusInterfaceNameToNetworkStatusMap, err := mapMultusInterfaceNameToNetworkStatus(networkStatusAnnotationValue)
+	networkNameToDeviceInfo, err := mapNetworkNameToDeviceInfo(
+		networks,
+		networkStatusAnnotationValue,
+		vmispec.FilterSRIOVInterfaces(interfaces),
+	)
 	if err != nil {
 		return nil, err
 	}
-	networkNameScheme := namescheme.CreateNetworkNameSchemeByPodNetworkStatus(networks, multusInterfaceNameToNetworkStatusMap)
 
 	networkPCIMap := map[string]string{}
 	for _, sriovIface := range vmispec.FilterSRIOVInterfaces(interfaces) {
-		multusInterfaceName := networkNameScheme[sriovIface.Name]
-		networkStatusEntry, exist := multusInterfaceNameToNetworkStatusMap[multusInterfaceName]
+		deviceInfo, exist := networkNameToDeviceInfo[sriovIface.Name]
```

Comment:
In continuation or Or's comment a few lines before.
I think you can do this now:
```go
for netName, deviceInfo := range networkNameToDeviceInfo {
    ...
}
```

Wherever `sriovIface.Name` appears, `netName` can be used.